assets/src/stories-editor/components/video-block-edit-with-preview.js

```diff
@@ -0,0 +1,246 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { getBlobByURL, isBlobURL } from '@wordpress/blob';
+import {
+	Component,
+	createRef,
+} from '@wordpress/element';
+import {
+	BaseControl,
+	Button,
+	IconButton,
+	PanelBody,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	InspectorControls,
+	MediaUpload,
+	MediaUploadCheck,
+	RichText,
+} from '@wordpress/block-editor';
+
+const ALLOWED_MEDIA_TYPES = [ 'video' ];
+const VIDEO_POSTER_ALLOWED_MEDIA_TYPES = [ 'image' ];
+
+/**
+ * Mainly forked from the Core Video block Edit component, but allows the <video> to play instead of being disabled.
+ *
+ * There are very few changes from the Core Video block's components.
+ * The main change is that in render(), the <video> is not wrapped in <Disabled>, so it can play.
+ *
+ * @class
+ */
+class VideoBlockEditWithPreview extends Component {
```

Comment:
Much of this is simply copied from the Video block's [edit.js](https://github.com/WordPress/gutenberg/blob/cb718ffdff57e35f05f4239ff03ceb476ae69a86/packages/block-library/src/video/edit.js).

Reply:
Can you amend the docblock to mention the removed player controls?

Reply:
Sure!

Reply:
Ah, thanks for amending the DocBlock.